pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -209,27 +209,36 @@ private void applyQueryOptions(QueryContext queryContext) {
 
     // Set group-by query options
     if (QueryContextUtils.isAggregationQuery(queryContext) && queryContext.getGroupByExpressions() != null) {
-
       // Set maxInitialResultHolderCapacity
-      queryContext.setMaxInitialResultHolderCapacity(_maxInitialResultHolderCapacity);
-
+      Integer initResultCap = QueryOptionsUtils.getMaxInitResultCap(queryOptions);
```

Comment:
Let's use the same way to set the arguments for readability, e.g. always do if check